File: cpp/leetcode/746. Min Cost Climbing Stairs.cpp
 int minCostClimbingStairs(vector<int>& cost) {
        vector<int>dp(cost.size()+2);        
        for(int i=cost.size()-1;i>=0;i--){
            dp[i]=min(dp[i+1],dp[i+2])+cost[i];
        }
        return min(dp[0],dp[1]);
    }

File: ruby/746-Min-Cost-Climbing-Stairs.rb
def min_cost_climbing_stairs(cost)
  return 0 if cost.empty?
  return cost.first if cost.length == 1

  second = 0
  first = cost[-1]
  (cost.length - 2).downto(0) do |idx|
    cost1 = cost[idx] + first 
    cost2 = cost[idx] + second
    tmp = cost1 > cost2 ? cost2 : cost1
    second = first
    first = tmp
  end
  first > second ? second : first
end
